content/blog/open-sourcing-packetrusher-a-5g-core-performance-tester.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Open-sourcing PacketRusher: A 5G Core performance tester"
+date: 2023-12-13T09:44:43.865Z
+author: Valentin D'Emmanuele
+authorimage: /img/profile.png
+thumbnailimage: /img/thumbnail.png
+disable: false
+tags:
+  - opensource
+  - 5g
+  - gnodeb
+  - ue
+  - simulator
+---
+In the fast-evolving landscape of 5G technology, the demand for robust and efficient testing tools has never been higher. Enter PacketRusher, a cutting-edge 5G Core Network performance testing tool. As the lead developer behind this groundbreaking project, I am thrilled to share the power and potential PacketRusher holds in revolutionizing the way we test and optimize 5G networks.
+
+## PacketRusher
+
+PacketRusher is a tool dedicated to the performance testing and automatic validation of 5G Core Networks. It tests a 5G Core Network via its external interfaces, as defined by the 3GPP : N1, N2 and N3 interfaces. The tool can emulate various types of User Equipment (eg. 5G phones or 5G-enabled IoT devices) and gNodeBs (eg. antennas in Radio Access Network), and it can validate and load-tests the 5G core network’s Control Plane functions using a set of pre-defined procedures on top of both New Generation Application Protocol (NGAP) and Non-access Stratum (NAS-5GS) 5G protocols. It enables users to tests 5G Core Networks with up to 100k simulated UEs simultaneously all from a simple Linux virtual machine, without the needs for expensive commercial tools, or 100k phones on a table ;-)
```

Comment:
In the second sentence in this paragraph, please remove the comma after "external interfaces".
"User Equipment" should be written as "user equipment" (that one I do know :-) ).
I recommend breaking out the third sentence into two separate sentences:
1) The tool can emulate various types of user equipment.... and gNodeBs (eg. antennas in Radio Access Network).
2) It can also validate and load test the 5G core network's control plane functions using a set of pre-defined...5G protocols.

Otherwise, it would be quite a long sentence to sift through. Please note where I changed "load-tests" to "load test" and "Control Plane" to "control plane".

In the final sentence, change "tests" to "test" so it reads:
"It enables users to test 5G Core Networks with up to...

Also, change "the needs" to "the need".

Reply:
Will do, thanks! One of the big issues I have when writting texts is that I seem to always write seemingly never-ending sentences or paragraph-sized sentences :-)